cadence/server/database.go

```diff
@@ -0,0 +1,93 @@
+// database.go
+// SQLite configuration and population
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/dhowden/tag"
+	"github.com/kenellorando/clog"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+func dbConfig() (newdb *sql.DB, err error) {
+	clog.Info("dbConfig", "Setting up the database.")
+	newdb, err = sql.Open("sqlite3", "/cadence/music-metadata.db")
+	if err != nil {
+		clog.Error("dbConfig", "Failed to open database file!", err)
+		return nil, err
+	}
+	_, err = newdb.Exec(`DROP TABLE IF EXISTS aria`)
```

Comment:
I had the random idea - Couldn't we get close to live filesystem polling if we split this file out into its own binary?

I suppose we'd need to then call both, but we could set this up to do a `CREATE VIRTUAL TABLE aria_tmp`, populate, then begin a transaction, `DROP TABLE IF EXISTS aria`, `ALTER TABLE aria_tmp RENAME TO aria`, commit, sort of thing - Then, even if you do a query during a population run, you should still get data back from the old table.

Then the populator binary would just sit in an infinite loop of

```go
db, err := dbConfig()
if err != nil {
    clog.Error(...)
} else {
    if error := dbPopulate(); error != nil {
        clog.Error(...)
    }
}
time.sleep(<ACCEPTABLE INTERVAL>)
```

It'd be inefficient as heck, but it'd get the "feature" of FS polling (plus or minus an `<ACCEPTABLE INTERVAL>`) without the work to make it work...

Reply:
I'll think about this. I was considering running a goroutine to monitor for filesystem changes in the target directory, then just drop the table completely and call `dbPopulate()` when it sees changes. Having a temporary table for that second it takes to populate sounds like the right thing to do.